.github/workflows/publish.yml

```diff
@@ -199,14 +199,15 @@ jobs:
         allowUpdates: true
         name: ${{ github.ref_name }}
         body: ${{ steps.changelog.outputs.changelog }}
+    
+    - name: Authenticate GCP
+      uses: 'google-github-actions/auth@v2'
+      with:
+        credentials_json: '${{ secrets.GCP_CREDENTIALS_JSON }}'
 
     - name: Setup Google Cloud SDK
-      uses: google-github-actions/setup-gcloud@v1
-      with:
-        version: 'latest'
-        project_id: ${{ secrets.GCP_PROJECT_ID }}
-        service_account_key: ${{ secrets.GCP_SA_KEY }}
+      uses: google-github-actions/setup-gcloud@v2
 
     - name: Upload to GCP
       run: |
-        gsutil cp micro-rdk-server-esp32-ota.bin gs://${{ secrets.GCP_BUCKET_NAME }}/${{ github.ref_name }}/
+        gcloud storage cp micro-rdk-server-esp32-ota.bin gs://${{ secrets.GCP_BUCKET_NAME }}/${{ github.ref_name }}/
```

Comment:
The RDK seems to use something else to do this, via `google-github-actions/upload-cloud-storage`: https://github.com/viamrobotics/rdk/blob/main/.github/workflows/staticbuild.yml#L127